File: step_by_step/step_07/c/2941.c
#include <stdio.h>
#include <string.h>

int main(void){
    char croatia_alphabet[8][4] = {"c=", "c-", "dz=", "d-", "lj", "nj", "s=", "z="};

    char input[101];
    scanf("%s", input);
    int input_len = strlen(input);
    int count = 0; // 알파벳의 개수
    int cr_count = 0; //크로아티아 알파벳 표현에 쓰인 char 개수
// 1. 입력 문자열 중 croatia_alphabet에 있는 알파벳들 count
    for(int num=0; num<8; num++){ // 각각의 croatia alphabet
        int alphabet_len;
        if(num==2)
            alphabet_len = 3; // "dz="일 경우, 길이가 3
        else
            alphabet_len = 2;

        // 2글자(혹은 3글자)씩 croatia_alphabet인지 확인
        for(int i=0; i<=input_len-alphabet_len; i++){
            int is_croatia = 0; // 0: false, 1: true
            for(int j=0; j<alphabet_len; j++){
                if(input[i+j] != croatia_alphabet[num][j])
                    break; // 'break'는 하나의 for문만 탈출
                if(j==alphabet_len-1){
                    is_croatia = 1;
                    //printf("%s\n", croatia_alphabet[num]);
                }
            }
            
            if(is_croatia){ // croatia_alphabet이면 실행
                if(num==2){ // "dz="일 경우, 3글자이고 "z=" 에서 중복되므로 예외 처리
                    cr_count+=3; // (+) 3글자
                    cr_count-=2; // (-) 2글자
                }
                else{
                    count++;
                    cr_count+=2;
                }
            }
        }
    }

// 2. 남은 알파벳들 count
    count += input_len - cr_count;
    printf("%d\n", count);
}

File: step_by_step/step_07/python/2941.py
# 크로아티아 알파벳

croatia_alphabet = ["c=", "c-", "dz=", "d-", "lj", "nj", "s=", "z="]

input_str = input()
count = 0   # 알파벳 개수
str_count = 0   # input_str 중, 크로아티아 알파벳 표현에 쓰인 문자 개수
# 1. 입력 문자열 중 croatia_alphabet에 있는 알파벳들 count
for cr_letter in croatia_alphabet:
    num_letter = input_str.count(cr_letter)
    count = count + num_letter
    if cr_letter == "dz=":  # 3글자이고, 나중 z= 에서 중복으로 세는 것 때문에 예외 처리
        count = count - num_letter
        str_count = str_count + 3*num_letter - 2*num_letter
    else:
        str_count = str_count + 2*num_letter

# 2. 남은 알파벳들 count
count = count + (len(input_str) - str_count)
print(count)

# 처음에는 알파벳 확인 후 삭제하는 방법을 시도했으나, "nljj" 경우로 오류